Jdbc analyzers state persistence

velramiir opened the pull request:
*Issue #35 *

*Description of changes:*
Add a JdbcStateProvider for the JdbcAnalyzers, works for memory and local disk

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

velramiir committed:
copy state provider class for jdbc, work in progress

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
copy state provider class for jdbc, work in progress

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
copy state provider class for jdbc, work in progress

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
copy state provider class for jdbc, work in progress

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
copy state provider class for jdbc, work in progress

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
copy state provider class for jdbc, work in progress

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
finish two state providers, in memory and local disc

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Merge branch 'jdbc-analyzers-state-persistence' of https://github.com/hpi-bp1819-naumann/deequ into jdbc-analyzers-state-persistence

velramiir committed:
fix frequencies persistence

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix frequencies persistence

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix frequencies persistence

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix frequencies persistence

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix frequencies persistence

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix wrong English in names, rename 'disc' to 'disk'

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
fix style violations

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
update pom.xml with input encoding utf-8 to fix style check on windows

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
update pom.xml with input encoding utf-8 to fix style check on windows

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
update pom.xml with input encoding utf-8 to fix style check on windows

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
update pom.xml with input encoding utf-8 to fix style check on windows

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Merge remote-tracking branch 'origin/jdbc-analyzers-state-persistence' into jdbc-analyzers-state-persistence

# Conflicts:
#	src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalHistogramWithJdbc.scala
#	src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcStateProvider.scala
#	src/main/scala/com/amazon/deequ/io/LocalDiskUtils.scala

phSch08 commented on src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalHistogramWithJdbc.scala:
```diff
@@ -0,0 +1,40 @@
+
+    println(analyzer.calculate(table))
+
+    val stateProvider = JdbcFileSystemStateProvider("C:\\Users\\justu\\Documents\\histo")
```
Please remove this

phSch08 commented on src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcStateProvider.scala:
```diff
@@ -0,0 +1,282 @@
+
+      case _: JdbcHistogram =>
+        persistFrequenciesLongState(state.asInstanceOf[JdbcFrequenciesAndNumRows], identifier)
+
```
You may add Entropy here

phSch08 commented on src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcStateProvider.scala:
```diff
@@ -0,0 +1,282 @@
+      case _ : JdbcMaximum => MaxState(loadDoubleState(identifier))
+
+      case _ : JdbcHistogram => loadFrequenciesLongState(identifier)
+
```
You may add Entropy here

velramiir committed:
add jdbc entropy to jdbc state provider and remove explicit path from
manual test case incrementalHistogramWithJdbc as requested in PR

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
add jdbc entropy to jdbc state provider and remove explicit path from
manual test case incrementalHistogramWithJdbc as requested in PR

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
add jdbc entropy to jdbc state provider and remove explicit path from
manual test case incrementalHistogramWithJdbc as requested in PR

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Merge branch 'jdbc-analyzers-state-persistence' of https://github.com/hpi-bp1819-naumann/deequ into jdbc-analyzers-state-persistence

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

# Conflicts:
#	src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalHistogramWithJdbc.scala
#	src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcStateProvider.scala
#	src/main/scala/com/amazon/deequ/io/LocalDiskUtils.scala

velramiir committed:
Fix style errors created by merge

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Fix style errors created by merge

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Fix style errors created by merge

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

phSch08 commented on src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalMaximumWithJdbc.scala:
```diff
@@ -0,0 +1,42 @@
+
+    val maxState = analyzer.computeStateFrom(table).getOrElse(MaxState(0))
+
+    val stateProvider = JdbcFileSystemStateProvider("")
```
```suggestion
    val stateProvider = JdbcFileSystemStateProvider("")// file system path to store at
```

phSch08 committed:
Update src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalMaximumWithJdbc.scala

add explanation to both files

Co-Authored-By: velramiir <33062216+velramiir@users.noreply.github.com>

phSch08 committed:
Update src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalMaximumWithJdbc.scala

add explanation to both files

Co-Authored-By: velramiir <33062216+velramiir@users.noreply.github.com>

velramiir committed:
Add tests for jdbc state provider and state aggregation tests for jdbc
analyzers

update Spark state aggregation tests with some more analyzers

add merge function to jdbc analyzers

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Add tests for jdbc state provider and state aggregation tests for jdbc
analyzers

update Spark state aggregation tests with some more analyzers

add merge function to jdbc analyzers

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
add new analyzers to jdbc state provider, jdbc provider tests pass now

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
add new analyzers to jdbc state provider, jdbc provider tests pass now

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Merge remote-tracking branch 'origin/jdbc-analyzers-state-persistence' into jdbc-analyzers-state-persistence

# Conflicts:
#	src/main/scala/com/amazon/deequ/analyzers/jdbc/IncrementalMaximumWithJdbc.scala
#	src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcStateProvider.scala
#	src/main/scala/com/amazon/deequ/io/LocalDiskUtils.scala

velramiir committed:
close result set in all analyzers to fix aggregation test

change sqlite database to in-memory db

add numeric columns to aggregation test data to be abale to test correlation

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
close result set in all analyzers to fix aggregation test

change sqlite database to in-memory db

add numeric columns to aggregation test data to be abale to test correlation

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

velramiir committed:
Merge remote-tracking branch 'origin/jdbc-analyzers-state-persistence' into jdbc-analyzers-state-persistence

# Conflicts:
#	src/main/scala/com/amazon/deequ/io/LocalDiskUtils.scala

velramiir committed:
fix scalastyle

Signed-off-by: velramiir <justus.zeinert@student.hpi.de>

PaulZGit commented on src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcCorrelation.scala:
```diff
@@ -103,32 +104,29 @@ case class JdbcCorrelation(firstColumn: String,
 
     val result = statement.executeQuery()
 
-    try {
```
Strange, why is try/catch missing here?

velramiir replied:
because it's not included in any analyzer

PaulZGit reviewed (commented):
Strange: Already made changes are now newly in this branch/removed from master
Maybe we should talk to our tutors about that :O

velramiir committed:
Merge branch 'jdbc-analyzers' into jdbc-analyzers-state-persistence
